vnext/local-cli/generator-windows/templates/cpp/src/App.cpp

```diff
@@ -38,3 +42,39 @@ App::App() noexcept
 
     InitializeComponent();
 }
+
+/// <summary>
+/// Invoked when the application is launched normally by the end user.  Other entry points
+/// will be used such as when the application is launched to open a specific file.
+/// </summary>
+/// <param name="e">Details about the launch request and process.</param>
+void App::OnLaunched(LaunchActivatedEventArgs const& e)
+{
+    using super = AppT<App>;
+    super::OnLaunched(e);
+
+    Frame rootFrame = Window::Current().Content().as<Frame>();
+    rootFrame.Navigate(xaml_typename<<%=name%>::MainPage>(), box_value(e.Arguments()));
+}
+
+/// <summary>
+/// Invoked when application execution is being suspended.  Application state is saved
+/// without knowing whether the application will be terminated or resumed with the contents
+/// of memory still intact.
+/// </summary>
+/// <param name="sender">The source of the suspend request.</param>
+/// <param name="e">Details about the suspend request.</param>
+void App::OnSuspending([[maybe_unused]] IInspectable const& sender, [[maybe_unused]] SuspendingEventArgs const& e)
+{
+    // Save application state and stop any background activity
+}
+
+/// <summary>
+/// Invoked when Navigation to a certain page fails
+/// </summary>
+/// <param name="sender">The Frame which failed navigation</param>
+/// <param name="e">Details about the navigation failure</param>
+void App::OnNavigationFailed(IInspectable const&, NavigationFailedEventArgs const& e)
+{
+    throw hresult_error(E_FAIL, hstring(L"Failed to load Page ") + e.SourcePageType().Name);
+}
```

Comment:
Could we avoid having these two methods in the default ReactNative template?
We already have a redbox and other facilities in RN for the lifetime management. I wonder if they bring any value.

Reply:
@vmoroz the redbox will show up when something in the RN side goes wrong. Page navigation failures can happen even without any RN content / before RN content loads,  e.g. if the page hits an exception loading XAML markup, etc. so I think there is still value in having these here, but let me know if you disagree